nbviewer/providers/gist/handlers.py

```diff
@@ -111,94 +129,139 @@ def get(self, user, gist_id, filename=''):
             return
 
         files = gist['files']
+
         many_files_gist = (len(files) > 1)
 
-        if not many_files_gist and not filename:
-            filename = list(files.keys())[0]
+        # user and gist_id get modified
+        return user, gist_id, gist, files, many_files_gist
 
-        if filename and filename in files:
-            file = files[filename]
-            if (file['type'] or '').startswith('image/'):
-                app_log.debug("Fetching raw image (%s) %s/%s: %s", file['type'], gist_id, filename, file['raw_url'])
-                response = yield self.fetch(file['raw_url'])
-                # use raw bytes for images:
-                content = response.body
-            elif file['truncated']:
-                app_log.debug("Gist %s/%s truncated, fetching %s", gist_id, filename, file['raw_url'])
-                response = yield self.fetch(file['raw_url'])
-                content = response_text(response, encoding='utf-8')
+   # Analogous to GitHubTreeHandler
+    @gen.coroutine
+    def tree_get(self, user, gist_id, gist, files):
+        """
+        provider_url:
+            URL to the notebook document upstream at the provider (e.g., GitHub)
+        executor_url: str, optional
+            URL to execute the notebook document (e.g., Binder)
```

Comment:
We should remove `provider_url` or `executor_url` from the docstring here and optionally document the actual parameters.